src/core/pointcloud/qgspointclouddataprovider.h

```diff
@@ -312,6 +348,14 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
     QString subsetString() const override;
     bool setSubsetString( const QString &subset, bool updateFeatureCount = false ) override;
 
+    /**
+     * Returns the statistics calculator object for the data provider
+     * The returned object can be used to increase the number of samples used in the statistcs calculation
+     *
+     * \since QGIS 3.26
+     */
+    QgsPointCloudStatsCalculator *statsCalculator() const;
```

Comment:
I don't think we should have this in provider API - the stats calculator object is an implementation detail in my opinion that should stay private...